src/metabase/search/appdb/specialization/mysql.clj

```diff
@@ -0,0 +1,73 @@
+(ns metabase.search.appdb.specialization.mysql
+  (:require
+   [clojure.math :as math]
+   [clojure.string :as str]
+   [honey.sql :as sql]
+   [metabase.search.appdb.specialization.api :as specialization]
+   [toucan2.core :as t2]))
+
+(defmethod specialization/table-schema :mysql [base-schema]
+  (into [[:id :bigint :auto-increment :primary-key]
+         [:search_text :text :not-null]
+         [:with_native_query_text :text :not-null]]
+        (->> base-schema
+             (map (fn [cdef] (if (some #{:timestamp-with-time-zone} cdef)
+                               (mapv #(if (= % :timestamp-with-time-zone) :timestamp %) cdef)
+                               cdef))))))
+
+(defmethod specialization/post-create-statements :mysql [prefix table-name]
+  (mapv
+   (fn [template] (format template prefix table-name))
+   ["ALTER TABLE %s ADD UNIQUE INDEX %s_identity_idx (model, model_id)"
+    "CREATE FULLTEXT INDEX %s_search_text_idx ON %s (search_text)"
+    "CREATE FULLTEXT INDEX %s_native_fulltext_idx ON %s (with_native_query_text)"
+    "CREATE INDEX %s_model_archived_idx ON %s (model, archived)"
+    "CREATE INDEX %s_archived_idx ON %s (archived)"]))
+
+(defmethod specialization/batch-upsert! :mysql [table entries]
+  (when (seq entries)
+    (let [update-keys (vec (disj (set (keys (first entries))) :id :model :model_id))]
```

Comment:
Oh man what a sad amount of transforming, especially considering this is only used with the further `into {}` transformation below. I'm too lazy to think through it all, but my intuition is that fusing the two transform together we can simplify it a lot. 

The overhead here is probably dominated by other things, but still bothers me 😄